Assets/Player/PlayerCamera.gd

```diff
@@ -68,7 +68,7 @@ func _unhandled_input(event: InputEvent) -> void:
 	active_context.interact(event, target_object, target_pos)
 
 func assign_to_player() -> Player:
-	return Global.Game.player if Global.Game != null and Global.Game.player else null
+	return Global.World.player if is_instance_valid(Global.World) and is_instance_valid(Global.World.player) else null
```

Comment:
What's the advantage of `is_instance_valid()` over `null` checks?

Reply:
Checking against `null` on freed objects is misleading:
https://www.reddit.com/r/godot/comments/s6g1mw/psa_question_use_is_instance_valid_instead_of/

Reply:
**Note for future refactoring:**
Once https://github.com/godotengine/godot/pull/73896 is merged, checking for validation can be expressed through booleaned objects (`if potentially_freed_object`) which serves as a shortcut for `if is_instance_valid(potentially_freed_object)` (https://github.com/godotengine/godot/pull/73896#issuecomment-1667852685).

Reply:
Yes, even though we'll need to remove `!= null` checks as they always return `true` eventually.

Reply:
Technically they do work in Godot 4 as in returning `true` on freed objects (reddit thread discusses Godot 3 behavior) but of course they still should be removed if it's the wrong approach or considered as a bug.

Reply:
Yes, they do work, but I think the major point is that freed objects aren't the same as `null`, mainly in hashing contexts (like dictionaries):
```gdscript
var node = Node.new()
node.free()

if hash(node) == hash(null):
	print("the hash is the same")
else:
	print("the hash is different")
```
So, if we cannot guarantee they will be the same thing in all contexts I don't think it's a reliable solution for checking object validity.